src/signTxOutput.c

```diff
@@ -468,10 +497,20 @@ static void signTxOutput_handleTokenAPDU(uint8_t* wireDataBuffer, size_t wireDat
 		read_view_t view = make_read_view(wireDataBuffer, wireDataBuffer + wireDataSize);
 
 		VALIDATE(view_remainingSize(&view) >= 4, ERR_INVALID_DATA);
-		token->assetNameSize = parse_u4be(&view);
+		const size_t newAssetNameSize = parse_u4be(&view);
+		PRINTF("  asst 1 >%.*H<\n", wireDataSize, wireDataBuffer);
+		PRINTF("  asst 2 >%d<\n", token->assetNameSize);
+		VALIDATE(ASSET_NAME_SIZE_MAX + 1 == token->assetNameSize || signTxOutput_lexiCompareViews(
+			make_read_view(token->assetNameBytes, token->assetNameBytes + token->assetNameSize),
+			make_read_view(wireDataBuffer + 4, wireDataBuffer + 4 + newAssetNameSize)
+			) < 0, ERR_INVALID_DATA);	//TODO should be STATE?
+		PRINTF("  asst 3\n");
```

Comment:
I try to avoid cryptic names (asst vs. asset).

Reply:
I don't plan to leave any of those prints in the final version.